### ISSUE TITLE
Page ordering - pages should be order by their creation date

### ISSUE THREAD

**Issue by malinajirka** [wordpress-mobile/WordPress-Android#10440]
### Expected behavior
Order of pages (My Site -> Site pages) should be identical to the order of pages presented in Calypso.

### Actual behavior
Pages in the app are ordered alphabetically as oppose to Calypso where they are order by their creation date (same as posts).

### Steps to reproduce the behavior
1. My Site
2. Site pages
3. Notice the pages are ordered alphabetically

**Comment by malinajirka** [comment]
@planarvoid If I recall correctly you worked on Properly Placed pages. Do you remember if the sorting is performed on the client or server side?
I believe we should do the sorting on the server side - not duplicate the "complex" logic on each client.

**Comment by planarvoid** [comment]
Before we start working on this. I'd like us to reconsider possible solutions. I don't think fallback to chronological list is the best user experience (I think the hierarchical POV is much more useful and users would just lose that forever). I have two suggestions. One is to make the hierarchy collapsable/expandable and collapse it by default if the user has too many pages. The second one is to add an ordering button to the toolbar so the user can sort the pages in any way they want. What do you think @osullivanchris ?

**Comment by osullivanchris** [comment]
Just noting that on iOS it is ordered by most recent like Calypso, so there's two points of reference there. 

> I think the hierarchical POV is much more useful
Testing this out on iOS where we have order by most recent, but once I add parent/child relationships, it no longer orders by recently updated, it is ordered by hierarchy once you have it.

> The second one is to add an ordering button to the toolbar
Because of the previous point I see that this can be useful as we have the power to order it different ways, but we're limiting ourselves to ordering it only one particular way at a time. I wonder whether ordering it would be per session/per device or set it everywhere? I think you are suggesting per session. That would be most practical if we're just doing it for the apps. I wonder if people would also want to create a custom view across devices/web to the order of the pages in a way that makes sense to them and the content of their site. Particularly if multiple people are working on a complex site with multiple pages, I feel like it might help people if they can organise it for their own needs?

> One is to make the hierarchy collapsable/expandable and collapse it by default if the user has too many pages.
This seems like a good idea to me. Probably a separate task as it would not to be done to solve the current discussion.

Some numbers on how many pages sites typically have would help to understand how big these issues are.

**Comment by planarvoid** [comment]
> per session/per device or set it everywhere

I think it should be `per user & site & device` because it allows users to set the view that works for them per site and in the mobile app. I think it is different on a smaller screen than on a large computer display. I think we should store the setting locally per site for the user. Different settings might work differently for other sites. 

Anything that would work across devices/web would be much more complex and would require work on the backend and API so that would be out of scope for this issue and we wouldn't be able to work on that now. 

The local ordering I was considering would be chronological/alphabetical, ASC/DESC, hierarchy turned on/off. The default would be alphabetical, ASC, hierarchy turned on. We could also simplify this to just two options ("latest first" and "A/Z" or something). 

> Probably a separate task as it would not to be done to solve the current discussion.

You're right. I thought it might help users with sites that have 100+ pages (where the hierarchy might be hard to understand).

**Comment by elibud** [comment]
I think we should be practical here and match Calypso and iOS on the Android app, and then make improvements based on user feedback, any objections?

**Comment by osullivanchris** [comment]
@elibud yeah that makes sense to me as a first step, and I think there are some nice ideas here which can be captured in separate improvement tickets.
